src/components/site-form.tsx

```diff
@@ -254,6 +278,50 @@ export const SiteForm = ( {
 					<span className="font-semibold">{ __( 'Site name' ) }</span>
 					<TextControlComponent onChange={ setSiteName } value={ siteName }></TextControlComponent>
 				</label>
+				{ setUseCustomDomain && setCustomDomain && (
+					<div className="flex items-center gap-2 mb-6">
+						<input
+							type="checkbox"
+							id="use-custom-domain"
+							checked={ useCustomDomain }
+							onChange={ ( e ) => {
+								setUseCustomDomain( e.target.checked );
+								setCustomDomain( generateDomainFromSiteName( siteName ) );
+							} }
+						/>
+						<label htmlFor="use-custom-domain">{ __( 'Use custom domain' ) }</label>
```

Comment:
Should we follow the current design from Figma, which assumes we place the 'Local domain' field in the 'Advance settings' block?